### PR TITLE
GH-10967 Migrate Post.PermanentDeleteBatch to Sync by default

### PR DESCRIPTION
**Pull request by lurcio**
#### Summary
Migrates 'Post.PermanentDeleteBatch' to Sync by default.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10967

### CODE REVIEW

**Comment by jespino** [comment]
This needs changes in the enterprise repo too, so it will take a bit longer to get merged, don't worry about the tests failing.

File: store/storetest/post_store.go [hunk 1]
```diff
@@ -2030,7 +2030,9 @@ func testPostStorePermanentDeleteBatch(t *testing.T, ss store.Store) {
 	o3.CreateAt = 100000
 	o3 = (<-ss.Post().Save(o3)).Data.(*model.Post)
 
-	store.Must(ss.Post().PermanentDeleteBatch(2000, 1000))
+	if _, err := ss.Post().PermanentDeleteBatch(2000, 1000); err != nil {
```
Comment (jespino) [inline store/storetest/post_store.go]: Nit: this is totally fine and consistent with the surrounding code but we are trying to use `require.Nil(t,err)` instead.

Comment (lurcio) [reply store/storetest/post_store.go]: Thank you for this @jespino, I've updated my PR to use the new style

**Comment by jespino** [comment]
@lurcio do you mind to merge master into this branch, I think everything is now ready to get merged :)

**Comment by lurcio** [comment]
@jespino Thank you for the update, I have merged master into my branch now.

**Comment by jespino** [comment]
@lurcio thanks! I'm working in merging the enterprise changes! so... soon your PR will be merged.

**Pull request merged**